Multi-Project-Extension

MysticJay opened the pull request:
After V0.30 was released the code, that was initially sumbmitted for PR#221 got corrupted.
Revamped the branch and fixed the code.

MysticJay committed:
Code updated for V0.30

After V0.30 was released the code, that was initially sumbmitted for PR#221 got corrupted.
Revamped the branch and fixed the code.

modos189 commented on plugins/bookmarks.js:
```diff
@@ -1,12 +1,17 @@
+// @id             Bookmarks
```
In our build system, if no ID is specified, the filename is used. Since the ID and filename match here, there is no point in specifying the ID

MysticJay replied:
will be removed

modos189 commented on plugins/bookmarks.js:
```diff
@@ -1,12 +1,17 @@
 // @category       Controls
-// @version        0.3.0
+// @version        0.4.0
 // @description    Save your favorite Maps and Portals and move the intel map with a click. Works with sync.
```
```suggestion
 // @description    Save your favorite Maps and Portals and move the intel map with a click. Works with sync. With support Multi Projects Extension.
```

MysticJay replied:
will be added

modos189 commented on plugins/draw-tools.js:
```diff
@@ -1,13 +1,21 @@
 // @category       Draw
-// @version        0.7.1
+// @version        0.8.0
 // @description    Allow drawing things onto the current map so you may plan your next move.
```
```suggestion
 // @description    Allow drawing things onto the current map so you may plan your next move. With support Multi Projects Extension.
```

MysticJay replied:
Will be changed

modos189 commented on plugins/bookmarks.js:
```diff
@@ -1,12 +1,17 @@
 
 
-/***********************************************************************
+// History ***********************************************
```
We're using git history, so please remove these lines

MysticJay replied:
will be removed

modos189 commented on plugins/bookmarks.js:
```diff
@@ -1303,4 +1352,11 @@
     if (window.plugin.portalslist) {
       window.plugin.bookmarks.setupPortalsList();
     }
+// Initilaize MPE-Support only if MPE-Module is available
```
```suggestion
    // Initilaize MPE-Support only if MPE-Module is available
```

MysticJay replied:
will be indented

modos189 commented on plugins/draw-tools.js:
```diff
@@ -198,15 +206,14 @@ window.plugin.drawTools.save = function() {
 window.plugin.drawTools.load = function() {
   try {
-    var dataStr = localStorage['plugin-draw-tools-layer'];
+        var dataStr = localStorage[window.plugin.drawTools.KEY_STORAGE];
```
```suggestion
    var dataStr = localStorage[window.plugin.drawTools.KEY_STORAGE];
```

MysticJay replied:
Indentation will be removed

modos189 commented on plugins/draw-tools.js:
```diff
@@ -363,10 +379,11 @@ window.plugin.drawTools.optCopy = function() {
         });
     }
 }
-
```
Apparently, a line was mistakenly deleted? It looks worse

MysticJay replied:
will be added

modos189 commented on plugins/draw-tools.js:
```diff
@@ -658,11 +675,95 @@ window.plugin.drawTools.boot = function() {
+  loadExternals();                              // initialize leaflet
+  window.plugin.drawTools.boot();               // initialize drawtools
+  window.pluginCreateHook('pluginDrawTools');
+  window.plugin.drawTools.setupCSS();           // create additional CSS elements
```
If this function is performed not only when MPE is active but always, maybe you should transfer the CSS code to a separate CSS file?

MysticJay replied:
Need to check, where this code is needed. Might be slipped in from upcomming features in DrawToolsPlus.
If kept it should move to the other CSS code in DT.

MysticJay replied:
The CSS changes are used to remove the GAP between drawitems and drawedit to safe space.
Maybe reintroduce at a different time.
line 759: setupCSS () removed.

MysticJay committed:
Sync only Default

Various smaller cosmetic changes
in Bookmarks only SYNC default Namespace

MysticJay committed:
Typos

BKMrks did not load
fixed typos

MysticJay committed:
fixed sync

BM
- highlights and markes follow the changed project.
- sync will only act on default storage.
- typos fixed
MPE
- removed debug code

modos189 committed:
Syntax correction

modos189 committed:
Delete font-awesome plugin

modos189 commented:
In the first commit I fixed the syntax, and in the second I removed the font-awesome. Why? In the task https://github.com/IITC-CE/ingress-intel-total-conversion/issues/236 we discussed the use of icons - they should be font or svg.
I'm in favor of using svg, but anyway - icons should not be in the form of a plugin, but in the core and become standard, otherwise only there two plugins will use icons and that will be the end of it.

Also note https://github.com/IITC-CE/ingress-intel-total-conversion/pull/312/files#diff-3073581708df23c3f1bd259521cb3021R8.
I replaced the code sample with a link to documentation - you just need to create this page and move the sample there

MysticJay commented:
Syntax changes are ok, while I prefer way more comments and explanations to each function. The dividers between the functions helped a lot to reverse engineer the code. But this is a different discussion.

FontAwesome: the current usage of SVGs already have negative side effects in some faction specific plugins, esp. if combined with "alternative text". But that is a different discussion too. 
FA is widely used in ZasoItems. I am fine keeping FA as a separate plugin with the other ZasoItems.

modos189 committed:
Merge branch 'master' of github.com:IITC-CE/ingress-intel-total-conversion into Multi-Project-Extension

 Conflicts:
	plugins/draw-tools.js
